.github/workflows/build-macos.yaml

```diff
@@ -0,0 +1,85 @@
+# Copyright 2025 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: MacOS Build and Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  contents: read
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: ${{ matrix.cxx }}, ${{ matrix.build_type }}
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        build_type: [RelWithDebugInfo]
+        cxx: [clang++-15]
```

Comment:
Can we include at least one gcc compilation as well (maybe gcc-11)

Reply:
Not with the current version :( GCC requires some additional tuning.

Reply:
I would say we should wait for GCC on mac until specific request. This would be quite a non standard setup.  Clang considered more native approach there.

Perf libs was supporting only clang on mac, back in the days

Reply:
Ok, sounds good to me